godot-core/src/obj/onready.rs

```diff
@@ -65,10 +75,77 @@ use std::mem;
 ///        assert_eq!(*self.manual, 22);
 ///     }
 /// }
+/// ```
+///
+/// # Example - Macro-generated `init`
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct MyClass {
+///    base: Base<Node>,
+///    #[init(default = OnReady::new(|| 11))]
+///    auto: OnReady<i32>,
+///    #[init(default = OnReady::manual())]
+///    manual: OnReady<i32>,
+///    #[init(node = "ChildPath")]
+///    auto_node: OnReady<Gd<Node2D>>,
+/// }
+///
+/// #[godot_api]
+/// impl INode for MyClass {
+///     fn ready(&mut self) {
+///        // self.auto is now ready with value 11.
+///        assert_eq!(*self.auto, 11);
+///
+///        // self.manual needs to be initialized manually.
+///        self.manual.init(22);
+///        assert_eq!(*self.manual, 22);
+///
+///        // self.auto_node is now ready with the node found at path `ChildPath`
+///        assert_eq!(self.auto_node.get_name(), "ChildPath".into());
+///     }
+/// }
+/// ```
 pub struct OnReady<T> {
     state: InitState<T>,
 }
 
+impl<T: GodotClass + Inherits<Node>> OnReady<Gd<T>> {
+    /// Variant of [OnReady::new], except this is hard-coded to fetch the node at `path` before ready.
+    ///
+    /// This is the functional equivalent of the GdScript pattern: `@onready var node = $NodePath`
+    ///
+    /// # Panics
+    /// - If `path` does not point to a valid node.
+    ///
+    /// Note that the panic will only happen if and when the node enters the SceneTree for the first time.
+    ///  (i.e.: it receives the `READY` notification)
+    ///
+    /// For a non-panicking version, see [OnReady<Option<Gd<T>>>::try_node]
+    pub fn node(path: impl Into<NodePath>) -> Self {
+        let path = path.into();
+        Self::from_base_fn(|base| base.get_node_as(path))
+    }
+}
+
+impl<T: GodotClass + Inherits<Node>> OnReady<Option<Gd<T>>> {
+    /// Variant of [OnReady::new], except this is hard-coded to fetch the node at `path` before ready.
+    ///
+    /// This is the functional equivalent of the GdScript pattern: `@onready var node = $NodePath`
+    ///
+    /// # Does not panic
+    /// If path does not point to a valid node, this will simply leave the field as `None`.
+    ///
+    /// If you're certain `path` is valid, and you're willing to panic if it isn't,
+    /// consider changing the field type to `OnReady<Gd<T>>`, then use [OnReady<Gd<T>>::node] instead.
+    pub fn try_node(path: impl Into<NodePath>) -> Self {
+        let path = path.into();
+        Self::from_base_fn(|base| base.try_get_node_as(path))
+    }
+}
```

Comment:
Out of curiosity, did you encounter a use cases where this was helpful?

Wondering because `OnReady` is generally used to have ergonomic access to late-initialized fields, and `Option` requires again explicit `.unwrap()`. In GDScript, `@onready var node = $NodePath` followed by `null` checks is not a typical pattern, or is it?

Reply:
@Houtamelo what are your thoughts on this? 🙂

Reply:
When writing that function, I had thought about a few use cases but after your question I re-considered it and realized the use cases were all redundant.